src/App.js

```diff
@@ -68,122 +46,142 @@ class App extends React.Component {
-    script.src = "https://cse.google.com/cse.js?cx=013104617978576650762:hrpvx9uejrs";
-    script.async = true;
-    document.body.appendChild(script);
-  }
+    // TODO: use get request to populate frontend
+    async function getRequest() {
+        const res = await fetch(`https://cjsback.herokuapp.com/`)
+        const data = await res.json()
+    }
 
-  querySearch(query) {
-    document.getElementById("gsc-i-id1").value = query;
-    const buttons = document.getElementsByClassName("gsc-search-button gsc-search-button-v2")
-    buttons[0].click();
-  }
+    // deals with the fact that setting state is asynchronous
+    useEffect(() => {
+        autoTranslate(input)
+    }, [input])
 
-  autoTranslate(input) {
-    this.allowTabs()
-    
-    // updates state with what user typed
-    this.setState({
-      input: input.target.value
-    })
+    async function postRequest(){
+        const params = new URLSearchParams({
+            input: input,
+            in_lang: inputLanguage,
+            out_lang: outputLanguage
+        })
 
-    clearTimeout(this.timeout)
-    this.timeout = setTimeout(() => {
-      // translates if 500 ms passes after last keystroke
-      this.postRequest()
-    }, 500);
-  }
-  
-  render() {
-    return (
-      <div>
-        <Navbar/>
-       <div class="selectlanguage">
-       <FormControl>
-        <InputLabel>Input</InputLabel>
-       <Select
-          value={this.inputLanguage}
-          defaultValue={"js"}
-          onChange={this.updateInputLanguage}
-        >
-          <MenuItem value={"js"}>JavaScript</MenuItem>
-          <MenuItem value={"py"}>Python</MenuItem>
-        </Select>
-        </FormControl>
-        
-       
-        <FormControl>
-        <InputLabel>Output</InputLabel>
-        <Select
-          value={this.outputLangauge}
-          defaultValue={"py"}
-          onChange={this.updateOutputLanguage}
-        >
-          <MenuItem value={"js"}>JavaScript</MenuItem>
-          <MenuItem value={"py"}>Python</MenuItem>
-        </Select>
-        </FormControl>
-        </div>
+        const requestOptions = {
+        method: 'POST',
+        headers: new Headers({
+            'Content-Type': 'application/x-www-form-urlencoded',
+        }),
+        body: params.toString()
+        };
+        const response = await fetch('https://cjsback.herokuapp.com/', requestOptions);    
+        const translation = await response.json();
 
+        setOutput(translation["response"])
+    }
+
+    const handleInputLanguageChange = (event, value) => {
+        setInputLanguage(value)
+    } 
+
+    const handleOutputLanguageChange = (event, value) => {
+        setOutputLanguage(value)
+    } 
+
+    function autoTranslate(value){
+        setInput(value);
+
+        clearTimeout(timer)
+        timer = setTimeout(() => {
+            postRequest()
+        }, 500)   
+    }
+
+    const handleInputChange = (event) => {
+        allowTabs();
```

Comment:
Feels odd to me to `allowTabs` in here, IMO would make more sense in the `useEffect` block.